spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -224,18 +222,14 @@ protected void afterUnbindProducer(ProducerDestination destination, P producerPr
 			bridge.setBeanName("bridge." + name);
 			consumerEndpoint = createConsumerEndpoint(destination, group, properties);
 			consumerEndpoint.setOutputChannel(bridge);
-			if (consumerEndpoint instanceof InitializingBean) {
```

Comment:
I'd like to understand the advantage of moving the lifecyle operations into `afterBind()`. I believe they, in fact, belong to `bind()` which already happens here.  

This makes me ask this question: Why do we need `bind()` while the actual binding is expected to happen during `doBindProducer` or `doBindConsumer`. I think we wanted to have `unbind()` to directly invoke the lifecycle `stop` operations and in the case of `bind()` we may not need an explicit method to do that as the binding operation is done  `doBindProducer` or `doBindConsumer`.